src/schema/order.js

```diff
@@ -270,29 +273,50 @@ export const resolvers = {
         };
       }
 
-      const agencyId = branch.agencyId;
+      // PickUpBranch agencyId
+      const agencyId = branch?.agencyId;
+
+      // userIds from userParameters
+      const userIds = getUserIds(args?.input?.userParameters);
 
       // Verify that the user is allowed to place an order
-      const { status, statusCode } = await getUserOrderAllowedStatus(
-        { agencyId },
+      const { status, statusCode, userId } = await getUserOrderAllowedStatus(
+        { agencyId, userIds },
         context
       );
 
       if (!status) {
         return { ok: status, status: statusCode };
       }
 
-      const input = {
-        ...args.input,
-        accessToken: context.accessToken,
-        smaug: context.smaug,
-        dryRun: args.dryRun,
-        branch,
-      };
+      // We assume we will get the verified userId from the 'getUserOrderAllowedStatus' check.
+      // If NOT (e.g. no borchk possible for agency), we fallback to an authenticated id and then an user provided id.
+      if (!userId && !context?.smaug?.user?.id && isEmpty(userIds)) {
+        // Order is not possible if no userId could be found or was provided for the user
+        return { ok: false, orderId: "fisk", status: "UNKNOWN_USER" };
```

Comment:
skal det måske bare være en tom string i stedet for fisk? :-)

Reply:
`const _userId = smaug.user.id || userId;`

Bruges til vores audit trace, her vil vi helst bruge id'et på den person som er logget ind med et auth token. Hvis brugeren ikke er logget ind, vil den smaug.user.id være tom, og så fallback'er den til det userId som vi har fået fra den er orderAllow funktion. 

fisk er fjernet :D